cloudify_agent/operations.py

```diff
@@ -446,3 +451,62 @@ def validate_agent_amqp(current_amqp=True, **_):
         raise NonRecoverableError(result['agent_alive_error'])
     if not current_amqp and not result['agent_alive_crossbroker']:
         raise NonRecoverableError(result['agent_alive_crossbroker_error'])
+
+
+@operation
+def transfer_agent_amqp(transfer_agent_timeout=300,
+                        manager_ip=None, manager_certificate=None,
+                        manager_rest_token=None, **_):
+
+    _create_broker_config(transfer_mode=True)
+    old_agent = _validate_agent()
+    agents = _run_install_script(old_agent, transfer_agent_timeout, manager_ip,
+                                 manager_certificate, manager_rest_token,
+                                 transfer_agent=True)
+    returned_agent = agents['new']
+    ctx.logger.info('Configured agent {0} to work with the new Manager'.
+                    format(returned_agent['name']))
+
+    # Make sure the agent is alive:
+    app = get_celery_app(tenant=returned_agent['rest_tenant'])
+    _assert_agent_alive(returned_agent['name'], app)
+
+
+def _create_broker_config(transfer_mode=False):
+    """
+    This function creates a dictionary called 'broker_config' within the
+    'cloudify_agent' dict, that contains all the required information that will
+     later be used to create a Celery client
+    :return:
+    """
+    if 'cloudify_agent' not in ctx.instance.runtime_properties:
+        raise NonRecoverableError(
+            'cloudify_agent key not available in runtime_properties')
+    agent = ctx.instance.runtime_properties['cloudify_agent']
+    agent['broker_config'] = dict()
```

Comment:
First check if it exists, I think. No need to recreate it on every validate command